tests/cross_functional/system_test/test_cluster_wide_key_rotation_system.py

```diff
@@ -0,0 +1,168 @@
+    verify_key_rotation_time,
+    enable_key_rotation,
+    set_key_rotation_time,
+    verify_new_key_after_rotation,
+)
+from ocs_ci.helpers.osd_resize import basic_resize_osd, get_storage_size
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs import constants, warp
+from ocs_ci.ocs.ocp import OCP
+
+log = logging.getLogger(__name__)
+
+
+@magenta_squad
+@system_test
+@ignore_leftovers
+class TestKeyRotationWithClusterFull(E2ETest):
+    """
+    1. Set cluster wide keyrotation period to every 5 mins by editing storage cluster and capture the keys details
+     for OSDs, Noobaa.
+    2.  Set custom taints to all the worker nodes and make sure #1 and #2 still exists after rebooting the pods.
+      a) Apply custom taint to all nodes
+      b)  Add toleration in  storagecluster CR and odf-operator subscription.
+      c) Verify toleration applied in ODF subscription and Storage Cluster CR are reflecting for other subscriptions,
+         Ceph and nooba components or not.
+      d) Verify the pods in all nodes are running as per taints applied.
+      e) Restart all ocs the pods on all nodes.
+
+    3. Fill the cluster till the full ratio limits (85%) by running IO from multiple pods and verify key rotation.
+    4. Once the cluster reaches read-only state, resize the OSD and verify key rotation.
+    5. Run noobaa_rebuild test and verify key rotation still works as it is.
+    6. Run noobaa-db backup and recovery locally. Verify key rotation.
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    def noobaa_rebuild_cleanup(self):
+        """
+        Cleanup function which clears all the noobaa rebuild entries.
+
+        """
+        # Get the deployment replica count
+        deploy_obj = OCP(
+            kind=constants.DEPLOYMENT,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        noobaa_deploy_obj = deploy_obj.get(
+            resource_name=constants.NOOBAA_OPERATOR_DEPLOYMENT
+        )
+        if noobaa_deploy_obj["spec"]["replicas"] != 1:
+            log.info(
+                f"Scaling back {constants.NOOBAA_OPERATOR_DEPLOYMENT} deployment to replica: 1"
+            )
+            deploy_obj.exec_oc_cmd(
+                f"scale deployment {constants.NOOBAA_OPERATOR_DEPLOYMENT} --replicas=1"
+            )
+
+    @pytest.fixture()
+    def warps3(self, request):
+        warps3 = warp.Warp()
+        warps3.create_resource_warp(replicas=4, multi_client=True)
+
+        def teardown():
+            warps3.cleanup()
+
+        request.addfinalizer(teardown)
+        return warps3
+
+    def test_cluster_wide_encryption_key_rotation_system(
+        self,
+        teardown_project_factory,
+        pvc_factory,
+        pod_factory,
+        threading_lock,
+        bucket_factory_session,
+        mcg_obj_session,
+        bucket_factory,
```

Comment:
Only use `bucket_factory_session` here since you're using `mcg_obj_session`. Using the function-scoped fixture will cause issues:

1. The function-scoped `bucket_factory` uses the function-scoped `mcg_obj`
2. The session scoped `bucket_factory_session` uses the session-scoped `mcg_obj_session`, which is a different class instant than `mcg_obj`
3. When you rebuild noobaa you get new credentials, but only `mcg_obj_session` gets them updated at `self.noobaa_rebuild_cleanup()` at row 143.
4. ==> `validate_noobaa_db_backup_recovery_locally_system` fails because you're using the function-scoped `bucket_factory` that uses the obsolete credentials

```suggestion
```